crypto/signatures_test.go

```diff
@@ -1,25 +1,141 @@
 package crypto
 
 import (
+	"bytes"
 	"crypto/rand"
+	"errors"
 	"testing"
 
 	"github.com/NebulousLabs/Sia/encoding"
+	"github.com/NebulousLabs/ed25519"
 )
 
-// Creates and encodes a public key, and verifies that it decodes correctly,
-// does the same with a signature.
-func TestSignatureEncoding(t *testing.T) {
+// mockKeyDeriver is a mock implementation of keyDeriver that saves its provided
+// entropy and allows the client to specify the returned SecretKey and
+// PublicKey.
+type mockKeyDeriver struct {
+	called  bool
+	entropy [EntropySize]byte
+	sk      ed25519.SecretKey
+	pk      ed25519.PublicKey
+}
+
+func (kd *mockKeyDeriver) deriveKeyPair(entropy [EntropySize]byte) (ed25519.SecretKey, ed25519.PublicKey) {
+	kd.called = true
+	kd.entropy = entropy
+	return kd.sk, kd.pk
+}
+
+// Test that the Generate method is properly calling its dependencies and
+// returning the expected key pair.
+func TestGenerateRandomKeyPair(t *testing.T) {
```

Comment:
Naming convention: `TestUnitXXX` for unit tests. 

You can run `make cover-unit` to see the code coverage for the unit tests, and `make cover-integration` to see the coverage for the integration tests. And `make cover` will show you for all of them.